Cython/Compiler/Nodes.py

```diff
@@ -4184,11 +4184,14 @@ def generate_tuple_and_keyword_parsing_code(self, args, code, decl_code):
         code.putln('{')
         all_args = tuple(positional_args) + tuple(kw_only_args)
         non_posonly_args = [arg for arg in all_args if not arg.pos_only]
-        non_pos_args_id = ','.join(
-            ['&%s' % code.intern_identifier(arg.entry.name) for arg in non_posonly_args] + ['0'])
-        code.putln("PyObject ** const %s[] = {%s};" % (
-            Naming.pykwdlist_cname,
-            non_pos_args_id))
+        if non_posonly_args:
+            non_pos_args_id = ','.join([
+                f'&{code.intern_identifier(arg.entry.name)}'
+                for arg in non_posonly_args
+            ] + ['0'])
+            code.putln("PyObject ** const %s[] = {%s};" % (
+                Naming.pykwdlist_cname,
+                non_pos_args_id))
```

Comment:
This could also use an f-string, just needs to double the `{` to get a literal.
```suggestion
            code.putln(f"PyObject ** const {Naming.pykwdlist_cname}[] = {{{non_pos_args_id}}};")
```

Reply:
Yeah, I had considered it but found the triple braces too easy to miss on a quick read and the `{%s}` to be clearer. I already hit a few cases where I found it sad that f-strings don't make the code look clearer, as in this case.

Not sure. Do you vote for it or just propose it as an alternative?

Reply:
Not important to me (and I don't pretend to have a vote :)), just thought I'd point it out.